### PR TITLE
refactor(compass-components): Add compass-components package, add Confirmation Modal

### PR DESCRIPTION
**Pull request by Anemy**
This PR adds a new package `compass-components`. This will be where we'll introduce new Compass components, wrap [leafygreen-ui](https://github.com/mongodb/leafygreen-ui)(LG) components, and overtime we'll move `hadron-react-components` into this package in the new format. 

Since this is some of the first typescript in Compass the pr is lighter on the implementation (just one components that's wrapping an LG component). What can be looked at and discussed more here is the typescript config, eslint config, testing, and file structure for these components. How would you like to see typescript packages look in Compass?

For testing we're using [react-testing library](https://testing-library.com/docs/react-testing-library/intro/) and [jest](https://jestjs.io/ ).

This doesn't introduce any styles yet, but we're planning on using [emotion](https://emotion.sh/docs/introduction) css in js. That will come in a future pr though and we can discuss it then. 

References:
- https://www.typescriptlang.org/ - Typescript
- https://jestjs.io/ - Jest testing framework
- https://reactjs.org/ - React
- https://testing-library.com/docs/react-testing-library/intro/ - React testing library
- https://github.com/mongodb/leafygreen-ui - LeafyGreen UI kit
- https://www.mongodb.design/ - MongoDB.Design

### CODE REVIEW

File: packages/compass-components/.eslintrc.js
Comment (mcasimir) [inline packages/compass-components/.eslintrc.js:6]: nice!

File: packages/compass-components/tsconfig.json
Comment (mcasimir) [inline packages/compass-components/tsconfig.json:1]: I'd start right away to put these in `ROOT/config/` as we do in mongosh and require these here already, so with the next package can just copy paste it.

File: packages/compass-components/package.json
Comment (addaleax) [inline packages/compass-components/package.json:18]:
```suggestion
    "compile": "npm run compile-ts && gen-esm-wrapper . ./lib/.esm-wrapper.mjs",
```

to align with the `"module"` entry above

we should probably also add an
```js
  "exports": {
    "require": "./lib/index.js",
    "import": "./lib/esm-wrapper.mjs"
  }
```

section to the package.json, I don’t think Node.js understands the `"module":` key

File: packages/compass-components/.eslintrc.js
Comment (mcasimir) [inline packages/compass-components/.eslintrc.js:11]: this is quite loose, would it be possible to make it consistent with the other products? in mongosh (https://github.com/mongodb-js/mongosh/blob/main/config/eslintrc.base.js) we also lint js somehow and we kept the style consistent with `eslint-config-mongodb-js` which will help a bit with the conversion and to keep things more aligned.

Also these overrides there makes lot of sense if they are not pulled in by default:

```
    '@typescript-eslint/no-unused-vars': 2,
    '@typescript-eslint/no-floating-promises': 2,
    '@typescript-eslint/await-thenable': 2,
    '@typescript-eslint/require-await': 2,
    '@typescript-eslint/explicit-module-boundary-types': 0,
    '@typescript-eslint/ban-types': 0,
    'mocha/no-skipped-tests': 1,
    'mocha/no-exclusive-tests': 2,
```

File: packages/compass-components/tsconfig.json
Comment (addaleax) [inline packages/compass-components/tsconfig.json:15]: I would also set this to `true`, if we can – it’s certainly helpful to have it enabled in mongosh

**Review by addaleax — APPROVED** [review]

File: packages/compass-components/.eslintrc.js
Comment (mcasimir) [inline packages/compass-components/.eslintrc.js:13]: What do we do for packages using `mocha`? Should we go with `jest` everywhere we convert to `ts`? Maybe we could pull this file in `ROOT/config` have a `eslintrc.base.js` and then a `ROOT/config/eslintrc.jest.js` and `ROOT/config/eslintrc.mocha.js` extending it with relevant section for both?

If everyone is ok with moving to `jest` we could (and maybe should) create some action for that to avoid having different technologies all over.  There seem to be a `jest-codemods` we could experiment around with.

File: packages/compass-components/jest.config.js
Comment (mcasimir) [inline packages/compass-components/jest.config.js:1]: I'd put this in the ROOT too, if that's the direction i think it doesn't hurt to just start that way

**Review by alenakhineika — APPROVED** [review]
Nice!

**Review by mcasimir — COMMENTED** [review]
It looks great.. looking forward to see lots components here!

Before merging it I'd would love to see linting rules aligned with the other products, and a bit less loose, as well centralised in one place.

Also we may want to have a strategy to move forward with jest. Not necessarily doing anything, but if we want to keep both for a while, we probably need a way to have linting rules / plugins that won't force only one, or vice-versa see if tests can be just converted.

File: packages/compass-components/.eslintrc.js
Comment (addaleax) [reply packages/compass-components/.eslintrc.js:11]: They’re not pulled in by default in mongosh, yup

File: packages/compass-components/.eslintrc.js
Comment (addaleax) [reply packages/compass-components/.eslintrc.js:13]: I mean, I like mocha a lot, but yeah, it would be nice to have consistency across Compass.

File: packages/compass-components/tsconfig.json
Comment (gribnoysup) [reply packages/compass-components/tsconfig.json:1]: FWIW I think it would be better to move them into a workspace inside the monorepo (like `packages/compass-config` or something), that way all config specific dependencies can stay in their own `package.json` instead of bundling up in the root one (same thing I think we should do for the root-level scripts)

File: README.md
Comment (gribnoysup) [inline README.md:18]: Probably should be added to the shared libraries and build tools section, not here

File: packages/compass-components/tsconfig.json
Comment (gribnoysup) [reply packages/compass-components/tsconfig.json:1]: But also personally I think not moving stuff right away to other places in this PR makes it easier to review and we can do a follow up that makes it available for other workspaces

File: packages/compass-components/tsconfig.json
Comment (gribnoysup) [inline packages/compass-components/tsconfig.json:17]: Having `strict` and `strictNullChecks` set to true is a bit redundant and can be omitted. Strict option [activates all `strict*` options](https://www.typescriptlang.org/tsconfig#strict)

File: packages/compass-components/tsconfig.json
Comment (gribnoysup) [inline packages/compass-components/tsconfig.json:1]: I like what's already here a lot, and would consider adding some more recommended options from typescript docs:

- [forceConsistentCasingInFileNames](https://www.typescriptlang.org/tsconfig#forceConsistentCasingInFileNames)
- [exactOptionalPropertyTypes](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes)
- [skipLibCheck](https://www.typescriptlang.org/tsconfig#skipLibCheck)

File: packages/compass-components/.eslintrc.js
Comment (rose-m) [reply packages/compass-components/.eslintrc.js:11]: `+1` on this one, I'd also prefer having it quite tight, especially also including linting rules for the tests, like `no-skipped` / `no-exclusive` things - they saved my butt a couple times already 👍

File: packages/compass-components/.eslintrc.js
Comment (rose-m) [reply packages/compass-components/.eslintrc.js:13]: I slightly prefer mocha tbh, but no real stakes in there. I especially found it easier to debug / develop individual tests with and had some pain in stubbing things in Jest. But as @mcasimir said we'd definitely have to agree that all migrated TS code then uses only `jest`.

**Review by rose-m — APPROVED** [review]
I really like it 🚀 Just added my supporting comments for both stricter linting and having a shared agreement on what to use for the tests for TS packages.

As @mcasimir mentioned I'd also be a fan of centralizing the configs (probably already in this PR) - either in a root dir or as @gribnoysup mentioned in a separate package, but to make sure a new TS package is essentially just copy-paste-adapt of the component package's files that refer to that centralized config.

File: packages/compass-components/.eslintrc.js
Comment (gribnoysup) [inline packages/compass-components/.eslintrc.js:10]: I think we can also add [react/recommended](https://github.com/yannickcr/eslint-plugin-react#list-of-supported-rules), [jsx-a11y/recommended](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y#rule-strictness-in-different-modes) and [jest/recommended](https://github.com/jest-community/eslint-plugin-jest#recommended) here right away

File: packages/compass-components/.eslintrc.js
Comment (gribnoysup) [reply packages/compass-components/.eslintrc.js:11]: I have been burned by poorly maintained custom configuration files for code linting and styling too much in the past, I _really_ want us to spend as little time as possible looking at this config in the future (without them being abandoned and getting outdated like `mongodb-js-eslint`), so I have a strong preference of maintaining as little of custom configurations as possible and would really like us not to include `mongodb-js-eslint` in any way in this new config (especially as we are planning to use it as a base for other packages in the monorepo) and in general would really love us to rely more on recommended configs and minimize the amount of rule overrides that we do in our configuration.

I totally get the concern that the recommended options might look quite loose especially as the first thing that you see in the configuration is just a few lines that extend from the recommended options, so I did a little comparison. I pulled a list of all rules that are currently used in mongosh + the ones suggested in https://github.com/mongodb-js/compass/pull/2368/files#r677604520 and diffed them with all the rules that eslint + typescript-eslint + react recommended would give us. At the first glance the difference is pretty big, mongosh config has 80+ rules that are missing from recommended ones, but from those 80:

- 28 are just style. FWIW eslint is not the best tool to maintain code style, but as I feel the general preference is to keep it like that, maybe we can take all those style rules and make _them_ into a separate shareable configuration and nothing else
- 6 are deprecated and should be replaced with other rules
- 2 are just redundant, don't make sense in combination with other rules. E.g., we have a `block-scoped-var` rule and a `no-var` rule at the same time
- 1 is literally a no-op because of how it is configured, lol

So we are left with ~40 rules out of ~140 that we probably should keep by just adding them directly to rules in this configuration (while maybe giving them another closer look, some of them don't look very useful to me tbh, things like `react/no-multi-comp` should definitely go away).

On top of that recommended rules have ~40 rules (none of them are style related 😉) that are not part of our current config, so all-in-all the recommended options give us more or less the same amount of checks, just different ones in my opinion. And I don't think it's a bad idea to incorporate those additional rules in our configuration anyway.

Here's an overview of all the rules that are part of our current configuration in mongosh:

❌ - redundant
🏚 - deprecated
👀 - totally let's keep them, but give them another thorough look
💅 - code style rules

||Type|Description|
|---|---|---|
|❌|[block-scoped-var](https://eslint.org/docs/rules/block-scoped-var)|Treat var as Block Scoped (block-scoped-var)|
|❌|[no-mixed-requires](https://eslint.org/docs/rules/no-mixed-requires)|disallow require calls to be mixed with regular variable declarations (no-mixed-requires)|
|🏚|[callback-return](https://eslint.org/docs/rules/callback-return)|Enforce Return After Callback (callback-return)|
|🏚|[handle-callback-err](https://eslint.org/docs/rules/handle-callback-err)|Enforce Callback Error Handling (handle-callback-err)|
|🏚|[no-negated-in-lhs](https://eslint.org/docs/rules/no-negated-in-lhs)|disallow negating the left operand in in expressions (no-negated-in-lhs)|
|🏚|[no-new-require](https://eslint.org/docs/rules/no-new-require)|Disallow new require (no-new-require)|
|🏚|[no-path-concat](https://eslint.org/docs/rules/no-path-concat)|Disallow string concatenation when using __dirname and __filename (no-path-concat)|
|🏚|[no-sync](https://eslint.org/docs/rules/no-sync)|Disallow Synchronous Methods (no-sync)|
|👀|[@typescript-eslint/await-thenable](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/await-thenable.md)|Disallows awaiting a value that is not a Thenable (await-thenable)|
|👀|[@typescript-eslint/no-floating-promises](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md)|Requires Promise-like values to be handled appropriately (no-floating-promises)|
|👀|[@typescript-eslint/require-await](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/require-await.md)|Disallow async functions which have no await expression (require-await)|
|👀|[complexity](https://eslint.org/docs/rules/complexity)|Limit Cyclomatic Complexity (complexity)|
|👀|[default-case](https://eslint.org/docs/rules/default-case)|Require Default Case in Switch Statements (default-case)|
|👀|[dot-notation](https://eslint.org/docs/rules/dot-notation)|Require Dot Notation (dot-notation)|
|👀|[eqeqeq](https://eslint.org/docs/rules/eqeqeq)|Require === and !== (eqeqeq)|
|👀|[guard-for-in](https://eslint.org/docs/rules/guard-for-in)|Require Guarding for-in (guard-for-in)|
|👀|[id-length](https://eslint.org/docs/rules/id-length)|enforce minimum and maximum identifier lengths (id-length)|
|👀|[new-cap](https://eslint.org/docs/rules/new-cap)|require constructor names to begin with a capital letter (new-cap)|
|👀|[no-alert](https://eslint.org/docs/rules/no-alert)|Disallow Use of Alert (no-alert)|
|👀|[no-caller](https://eslint.org/docs/rules/no-caller)|Disallow Use of caller/callee (no-caller)|
|👀|[no-console](https://eslint.org/docs/rules/no-console)|disallow the use of console (no-console)|
|👀|[no-else-return](https://eslint.org/docs/rules/no-else-return)|Disallow return before else (no-else-return)|
|👀|[no-eval](https://eslint.org/docs/rules/no-eval)|Disallow eval() (no-eval)|
|👀|[no-extend-native](https://eslint.org/docs/rules/no-extend-native)|Disallow Extending of Native Objects (no-extend-native)|
|👀|[no-extra-bind](https://eslint.org/docs/rules/no-extra-bind)|Disallow unnecessary function binding (no-extra-bind)|
|👀|[no-implied-eval](https://eslint.org/docs/rules/no-implied-eval)|Disallow Implied eval() (no-implied-eval)|
|👀|[no-iterator](https://eslint.org/docs/rules/no-iterator)|Disallow Iterator (no-iterator)|
|👀|[no-labels](https://eslint.org/docs/rules/no-labels)|Disallow Labeled Statements (no-labels)|
|👀|[no-lone-blocks](https://eslint.org/docs/rules/no-lone-blocks)|Disallow Unnecessary Nested Blocks (no-lone-blocks)|
|👀|[no-lonely-if](https://eslint.org/docs/rules/no-lonely-if)|disallow if statements as the only statement in else blocks (no-lonely-if)|
|👀|[no-loop-func](https://eslint.org/docs/rules/no-loop-func)|Disallow Functions in Loops (no-loop-func)|
|👀|[no-multi-str](https://eslint.org/docs/rules/no-multi-str)|Disallow Multiline Strings (no-multi-str)|
|👀|[no-native-reassign](https://eslint.org/docs/rules/no-native-reassign)|Disallow Reassignment of Native Objects (no-native-reassign)|
|👀|[no-nested-ternary](https://eslint.org/docs/rules/no-nested-ternary)|disallow nested ternary expressions (no-nested-ternary)|
|👀|[no-new](https://eslint.org/docs/rules/no-new)|Disallow new For Side Effects (no-new)|
|👀|[no-new-func](https://eslint.org/docs/rules/no-new-func)|Disallow Function Constructor (no-new-func)|
|👀|[no-new-object](https://eslint.org/docs/rules/no-new-object)|disallow Object constructors (no-new-object)|
|👀|[no-new-wrappers](https://eslint.org/docs/rules/no-new-wrappers)|Disallow Primitive Wrapper Instances (no-new-wrappers)|
|👀|[no-octal-escape](https://eslint.org/docs/rules/no-octal-escape)|disallow octal escape sequences in string literals (no-octal-escape)|
|👀|[no-proto](https://eslint.org/docs/rules/no-proto)|Disallow Use of __proto__ (no-proto)|
|👀|[no-return-assign](https://eslint.org/docs/rules/no-return-assign)|Disallow Assignment in return Statement (no-return-assign)|
|👀|[no-script-url](https://eslint.org/docs/rules/no-script-url)|Disallow Script URLs (no-script-url)|
|👀|[no-self-compare](https://eslint.org/docs/rules/no-self-compare)|Disallow Self Compare (no-self-compare)|
|👀|[no-sequences](https://eslint.org/docs/rules/no-sequences)|Disallow Use of the Comma Operator (no-sequences)|
|👀|[no-throw-literal](https://eslint.org/docs/rules/no-throw-literal)|Restrict what can be thrown as an exception (no-throw-literal)|
|👀|[no-unused-expressions](https://eslint.org/docs/rules/no-unused-expressions)|Disallow Unused Expressions (no-unused-expressions)|
|👀|[no-var](https://eslint.org/docs/rules/no-var)|require let or const instead of var (no-var)|
|👀|[one-var](https://eslint.org/docs/rules/one-var)|enforce variables to be declared either together or separately in functions (one-var)|
|👀|[prefer-const](https://eslint.org/docs/rules/prefer-const)|Suggest using const (prefer-const)|
|👀|[radix](https://eslint.org/docs/rules/radix)|Require Radix Parameter (radix)|
|👀|[react/jsx-boolean-value](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md)|Enforce boolean attributes notation in JSX|
|👀|[react/no-did-mount-set-state](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-mount-set-state.md)|Prevent usage of setState in componentDidMount|
|👀|[react/no-did-update-set-state](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-update-set-state.md)|Prevent usage of setState in componentDidUpdate (react/no-did-update-set-state)|
|👀|[react/no-multi-comp](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-multi-comp.md)|Prevent multiple component definition per file (react/no-multi-comp)|
|👀|[react/self-closing-comp](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/self-closing-comp.md)|Prevent extra closing tags for components without children (react/self-closing-comp)|
|👀|[strict](https://eslint.org/docs/rules/strict)|require or disallow strict mode directives (strict)|
|👀|[yoda](https://eslint.org/docs/rules/yoda)|Require or disallow Yoda Conditions (yoda)|
|💅|[@typescript-eslint/semi](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/semi.md)|Require or disallow semicolons instead of ASI (semi)|
|💅|[brace-style](https://eslint.org/docs/rules/brace-style)|Require Brace Style (brace-style)|
|💅|[camelcase](https://eslint.org/docs/rules/camelcase)|Require CamelCase (camelcase)|
|💅|[comma-dangle](https://eslint.org/docs/rules/comma-dangle)|require or disallow trailing commas (comma-dangle)|
|💅|[comma-spacing](https://eslint.org/docs/rules/comma-spacing)|Enforces spacing around commas (comma-spacing)|
|💅|[comma-style](https://eslint.org/docs/rules/comma-style)|Comma style (comma-style)|
|💅|[curly](https://eslint.org/docs/rules/curly)|Require Following Curly Brace Conventions (curly)|
|💅|[eol-last](https://eslint.org/docs/rules/eol-last)|require or disallow newline at the end of files (eol-last)|
|💅|[indent](https://eslint.org/docs/rules/indent)|enforce consistent indentation (indent)|
|💅|[jsx-quotes](https://eslint.org/docs/rules/jsx-quotes)|enforce the consistent use of either double or single quotes in JSX attributes (jsx-quotes)|
|💅|[key-spacing](https://eslint.org/docs/rules/key-spacing)|enforce consistent spacing between keys and values in object literal properties (key-spacing)|
|💅|[keyword-spacing](https://eslint.org/docs/rules/keyword-spacing)|enforce consistent spacing before and after keywords (keyword-spacing)|
|💅|[no-extra-parens](https://eslint.org/docs/rules/no-extra-parens)|disallow unnecessary parentheses (no-extra-parens)|
|💅|[no-floating-decimal](https://eslint.org/docs/rules/no-floating-decimal)|Disallow Floating Decimals (no-floating-decimal)|
|💅|[no-multi-spaces](https://eslint.org/docs/rules/no-multi-spaces)|Disallow multiple spaces (no-multi-spaces)|
|💅|[no-multiple-empty-lines](https://eslint.org/docs/rules/no-multiple-empty-lines)|disallow multiple empty lines (no-multiple-empty-lines)|
|💅|[no-spaced-func](https://eslint.org/docs/rules/no-spaced-func)|disallow spacing between function identifiers and their applications (no-spaced-func)|
|💅|[no-trailing-spaces](https://eslint.org/docs/rules/no-trailing-spaces)|disallow trailing whitespace at the end of lines (no-trailing-spaces)|
|💅|[object-curly-spacing](https://eslint.org/docs/rules/object-curly-spacing)|enforce consistent spacing inside braces (object-curly-spacing)|
|💅|[padded-blocks](https://eslint.org/docs/rules/padded-blocks)|require or disallow padding within blocks (padded-blocks)|
|💅|[quotes](https://eslint.org/docs/rules/quotes)|enforce the consistent use of either backticks, double, or single quotes (quotes)|
|💅|[react/jsx-wrap-multilines](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-wrap-multilines.md)|Prevent missing parentheses around multiline JSX (react/jsx-wrap-multilines)|
|💅|[semi-spacing](https://eslint.org/docs/rules/semi-spacing)|Enforce spacing before and after semicolons (semi-spacing)|
|💅|[space-before-blocks](https://eslint.org/docs/rules/space-before-blocks)|Require Or Disallow Space Before Blocks (space-before-blocks)|
|💅|[space-before-function-paren](https://eslint.org/docs/rules/space-before-function-paren)|Require or disallow a space before function parenthesis (space-before-function-paren)|
|💅|[space-infix-ops](https://eslint.org/docs/rules/space-infix-ops)|require spacing around infix operators (space-infix-ops)|
|💅|[spaced-comment](https://eslint.org/docs/rules/spaced-comment)|Requires or disallows a whitespace (space or tab) beginning a comment (spaced-comment)|
|💅|[wrap-iife](https://eslint.org/docs/rules/wrap-iife)|Require IIFEs to be Wrapped (wrap-iife)|

File: packages/compass-components/.eslintrc.js
Comment (gribnoysup) [reply packages/compass-components/.eslintrc.js:13]: Don't have a preference, from experience `jest` usually has more react-specific integrations available, e.g., `jsdom` env is included, tools like `testing-library` or `react-axe` usually come with a bunch of jest helpers already, whereas with `mocha` you'd usually need to configure it all yourself (but that's not really too critical, just a bit more maintenance on our side)

File: packages/compass-components/tsconfig.json
Comment (mcasimir) [reply packages/compass-components/tsconfig.json:1]: As long as we don't start other packages with different configurations that would be ok

File: packages/compass-components/.eslintrc.js
Comment (mcasimir) [reply packages/compass-components/.eslintrc.js:11]: Thanks for this, is super detailed @gribnoysup!

> So we are left with ~40 rules out of ~140 that we probably should keep by just adding them directly to rules in this configuration 

I think some of the one you have listed there may actually even be in the defaults .. maybe they are overridden in the `eslint-config-mongodb-js`? I'm sure stuff like `complexity` and  `quotes` comes for free.

I don't think that we have to use the `eslint-config-mongodb-js` unless that helps in any way. The defaults here in the PR are lacking few things though that are important practically speaking:
- don't forget await
- don't forget it.only on tests
- some rules for dependencies (some stuff are not detected by depcheck it seems), i think is some kind of recommended here: https://www.npmjs.com/package/eslint-plugin-node

Also I would love if we lint any js file (js and ts) also in the root / outside of plugins.

File: packages/compass-components/.eslintrc.js
Comment (mcasimir) [reply packages/compass-components/.eslintrc.js:13]: Would be great to make a decision though .. I also don't have a preference.

File: packages/compass-components/.eslintrc.js
Comment (mcasimir) [reply packages/compass-components/.eslintrc.js:10]: nowadays we could even "tailor" stuff to the file extension:

```
".js" --> only js rule (no typescript parser)
".ts !.spec.ts" --> ts parser, + typescript/recommended + node/recommended 
".tsx" -->  react/recommended, jsx-a11y/recommended 
"spec.ts" -->  jest/recommended 
```

or something similar.. is a relatively new feature of eslint

File: packages/compass-components/.eslintrc.js
Comment (gribnoysup) [reply packages/compass-components/.eslintrc.js:13]: So just to outline:

`mocha`:

\+ preferred by more people in the team (or at least seems like it)
\+ we don't need to drastically rewrite other tests in the monorepo
\- a bit more work to set up in some cases

`jest`:

\+ better react integrations
\- nobody have a preference specifically for it
\- we will need to gradually migrate other tests to it, so additional work

seems like we have a clear winner here 😄

File: packages/compass-components/.eslintrc.js
Comment (gribnoysup) [reply packages/compass-components/.eslintrc.js:11]: > I'm sure stuff like `complexity` and `quotes` comes for free.

No, they are not part of recommended, `quotes` are style related and eslint doesn't put anything from that group there and for `complexity` I don't know for sure, but I'd guess that as the number in configuration is debatable it would be hard to put in recommended in a way that would be equally helpful for everyone (and recommended options usually try to avoid this kind of situation)

File: packages/compass-components/.eslintrc.js
Comment (Anemy) [reply packages/compass-components/.eslintrc.js:18]: Updated the tests to use mocha. 👍  Definitely fewer react integrations but we can work around that.

File: packages/compass-components/tsconfig.json
Comment (Anemy) [reply packages/compass-components/tsconfig.json:1]: Holding off for now, let's be sure to move it into a centralized place then when we're adding another package.

File: packages/compass-components/.eslintrc.js
Comment (Anemy) [reply packages/compass-components/.eslintrc.js:16]: Updated the lint rules, now we're using recommended rules from most of the plugins. Looking through the mentioned rules it seems to grab all of them. `plugin:@typescript-eslint/recommended`, `plugin:mocha/recommended`, etc

**Pull request merged**
